src/CQRS/LeanCode.CQRS.Execution/CQRSObjectMetadata.cs

```diff
@@ -16,6 +16,32 @@ public CQRSObjectMetadata(CQRSObjectKind objectKind, Type objectType, Type resul
     }
 }
 
+public class CQRSObjectMetadataEqualityComparer : IEqualityComparer<CQRSObjectMetadata>
+{
+    public bool Equals(CQRSObjectMetadata? x, CQRSObjectMetadata? y)
+    {
+        if (ReferenceEquals(x, y))
+        {
+            return true;
+        }
+
+        if (x is null || y is null)
+        {
+            return false;
+        }
+
+        return x.ObjectKind == y.ObjectKind
+            && x.ObjectType == y.ObjectType
+            && x.ResultType == y.ResultType
+            && x.HandlerType == y.HandlerType;
+    }
+
+    public int GetHashCode(CQRSObjectMetadata obj)
+    {
+        return (obj.ObjectKind, obj.ObjectType, obj.ResultType, obj.HandlerType).GetHashCode();
```

Comment:
```suggestion
        return HashCode.Combine(obj.ObjectKind, obj.ObjectType, obj.ResultType, obj.HandlerType);
```

Maybe `CQRSObjectMetadata` should have been a `record class` instead?